src/core/openthread.cpp

```diff
@@ -66,9 +66,6 @@ static otDEFINE_ALIGNED_VAR(sInstanceRaw, sizeof(otInstance), uint64_t);
 otInstance *sInstance = NULL;
 #endif
 
-void OT_CDECL operator delete(void *, size_t) throw() { }
-void OT_CDECL operator delete(void *) throw() { }
```

Comment:
@mszczodrak, the windows build currently maps mbedtls `calloc()` and `free()` methods to dynamically allocate memory as needed by mbedtls.  The `Dtls` class in `src/core/meshcop/dtls.hpp` currently relies on the destructor to free all allocated memory as a part of `otInstanceFinalize()`.

If we really want to avoid the `delete` operator, an alternative maybe to define a new method that is simply called within `otInstanceFinalize()` and not rely on C++ destructors.  Thoughts?